vpr/src/place/static_move_generator.cpp

```diff
@@ -0,0 +1,58 @@
+#include "static_move_generator.h"
+#include "globals.h"
+#include <algorithm>
+
+#include "vtr_random.h"
+#include "vtr_assert.h"
+
+StaticMoveGenerator::StaticMoveGenerator(const std::vector<float>& prob) {
+    std::unique_ptr<MoveGenerator> uniform_move_generator;
+    uniform_move_generator = std::make_unique<UniformMoveGenerator>();
+    avail_moves.push_back(std::move(uniform_move_generator));
+
+    std::unique_ptr<MoveGenerator> median_move_generator;
+    median_move_generator = std::make_unique<MedianMoveGenerator>();
+    avail_moves.push_back(std::move(median_move_generator));
+
+    std::unique_ptr<MoveGenerator> weighted_centroid_move_generator;
+    weighted_centroid_move_generator = std::make_unique<WeightedCentroidMoveGenerator>();
+    avail_moves.push_back(std::move(weighted_centroid_move_generator));
+
+    std::unique_ptr<MoveGenerator> centroid_move_generator;
+    centroid_move_generator = std::make_unique<CentroidMoveGenerator>();
+    avail_moves.push_back(std::move(centroid_move_generator));
+
+    std::unique_ptr<MoveGenerator> weighted_median_move_generator;
+    weighted_median_move_generator = std::make_unique<WeightedMedianMoveGenerator>();
+    avail_moves.push_back(std::move(weighted_median_move_generator));
+
+    std::unique_ptr<MoveGenerator> crit_uniform_move_generator;
+    crit_uniform_move_generator = std::make_unique<CriticalUniformMoveGenerator>();
+    avail_moves.push_back(std::move(crit_uniform_move_generator));
+
+    std::unique_ptr<MoveGenerator> feasible_region_move_generator;
+    feasible_region_move_generator = std::make_unique<FeasibleRegionMoveGenerator>();
+    avail_moves.push_back(std::move(feasible_region_move_generator));
+
+    cumm_move_probs.push_back(prob[0]);
```

Comment:
This (lines 37-42) would be more clear and easier to test as a separate function.

Reply:
done by adding a new function to initialize the move_prob vector